dcpy/test/lifecycle/assemble/test_assemble.py

```diff
@@ -31,3 +35,41 @@ def test_unpack_package(metadata, tmp_path, package_path):
     # Check that files are unpacked correctly
     assert (dataset_files_path / "data_file.xlsx").exists()
     assert (attachments_path / "readme.pdf").exists()
+
+
+@pytest.fixture
+def colp_package_path(resources_path: Path):
+    return resources_path / "product_metadata" / "colp_single_feature_package"
+
+
+@mock.patch("dcpy.lifecycle.distribute.bytes.pull_destination_files")
+def test_assemble_from_bytes(pull_destination_files_mock, tmp_path, colp_package_path):
+    OTI_DD_FILENAME = "oti_dd.xlsx"
+    MOCK_PULLED_PACKAGE_PATH = tmp_path
+
+    pull_destination_files_mock.side_effect = lambda *args, **kwargs: shutil.copytree(
+        colp_package_path, MOCK_PULLED_PACKAGE_PATH, dirs_exist_ok=True
+    )
+
+    # Add an oti_xlsx file type, which we'll then have to generate
+    metadata = md.Metadata.from_path(
+        colp_package_path / "metadata.yml", template_vars={"version": "24b"}
+    )
+    metadata.files.append(
+        md.FileAndOverrides(
+            file=md.File(
+                id="oti_xlsx", filename=OTI_DD_FILENAME, type=OTI_METADATA_FILE_TYPE
+            )
+        )
+    )
```

Comment:
would adding a file with type `oti_data_dictionary` to `colp_package_path / "metadata.yml"` effectively do the same thing as this append?

I'd understand not wanting to modify it to reflect every possible test though

Reply:
yeah, that's a good (and open) question... I'd initially started by adding the OTI file there, but `dcpy/test/lifecycle/package/test_packages_validation.py` is using the COLP metadata package, and it tests validating the full package. So since that file is missing (ie not generated yet) it'll cause another test to fail. (TLDR: wanted to leave that as a valid/complete package)